packages/core/src/components/image-overlay/image-overlay.tsx

```diff
@@ -102,40 +115,78 @@ export class ImageOverlay implements ComponentInterface {
     this.zoomable = width < naturalWidth || height < naturalHeight;
   }
 
+  private isWijzigactie(wijzigActie: string): wijzigActie is ImageOverlayWijzigactie {
+    return wijzigActie === "voegtoe" || wijzigActie === "verwijder";
+  }
+
   render() {
     const { src, alt } = this.host.querySelector("img") ?? {};
 
+    const editActionLabel: string | undefined =
+      (this.wijzigactie && this.isWijzigactie(this.wijzigactie) && wijzigactieLabels[this.wijzigactie]) || undefined;
+
+    const dimmer = this.active && src && alt && (
```

Comment:
Ik begrijp het, maar dit is niet de JSX way™️.

Hier zouden we normaal een FunctionalComponent voor maken:

```jsx
const Dimmer: FunctionalComponent<{ active: boolean }> => ({ active }) => active && <div>active!</div>;

return <Dimmer active={this.active} />;
```

Je zou nog nog een lans kunnen breken dat het tonen van `<Dimmer />`  geen onderdeel van `<Dimmer />` is maar bij op de afnemende locatie komt te liggen, maar ik ben benieuwd naar jouw idee.

Omdat het een Functional Component (functie) is is je code ook portable/herbruikbaar by design. JSX nodes mogen niet herbruikt worden, de functie voorkomt dat.

Ik zou de Functional Component naar de root van de file verplaatsen (naast/onder de class). Hiermee dwing je ook nog eens af dat het een Pure component is.